src/Compilers/CSharp/Portable/FlowAnalysis/DefiniteAssignment.VariableIdentifier.cs

```diff
@@ -26,26 +27,60 @@ public bool Exists
             {
                 get { return (object)Symbol != null; }
             }
+
             public override int GetHashCode()
             {
-                return Roslyn.Utilities.Hash.Combine(Symbol?.OriginalDefinition, ContainingSlot);
+                Debug.Assert(Exists);
+
+                int currentKey = ContainingSlot;
+                // MemberIndexOpt, if available, is a cheap approach to comparing relative members.
```

Comment:
>MemberIndexOpt, if available, is a cheap approach to comparing relative members. [](start = 19, length = 80)

This makes it sound like it's just an optimization, rather than serving some functional purpose. Is that the case?